dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java

```diff
@@ -128,9 +128,36 @@ public static SourceCodeRepoInterface createSourceCodeRepo(String gitUrl, String
      * @param url
      * @return a map with keys: Source, Username, Repository
      */
-    public static Map<String, String> parseGitUrl(String url) {
-        Matcher m1 = GITHUB_REGEX_PATTERN_1.matcher(url);
-        Matcher m2 = GITHUB_REGEX_PATTERN_2.matcher(url);
+    public static Optional<Map<String, String>> parseGitUrl(String url) {
```

Comment:
With all the changes you've made for this, changing the signature to `Optional`, introducing contstants for the keys, you could go one step further and just introduce a type:

```
public class ParsedGitUrl {
  private String source;
  private String username;
  private String repository;
  ...
}

public static Optional<ParsedGitHurl> parseGitUrl(String url) {
```

But it's going way beyond the scope of the PR, so...

Reply:
Left it for another PR